scopes/mdx/mdx/mdx.main.runtime.ts

```diff
@@ -89,23 +91,27 @@ export class MDXMain {
         tsCompiler,
       ],
       {}
-    );
-    const mdxEnv = envs.compose(react.reactEnv, [
-      react.overrideCompiler(mdxCompiler),
-      react.overrideDependencies({
-        dependencies: {
-          '@teambit/mdx.ui.mdx-scope-context': '0.0.368',
-          '@mdx-js/react': '1.6.22',
-        },
-      }),
-      react.overrideCompilerTasks([compiler.createTask('MDXCompiler', mdxCompiler)]),
-    ]);
-    envs.registerEnv(mdxEnv);
-    depResolver.registerDetector(new MDXDependencyDetector(config.extensions));
-    docs.registerDocReader(new MDXDocReader(config.extensions));
-    generator.registerComponentTemplate(componentTemplates);
+      );
+      const envOverrides = [
+        react.overrideCompiler(mdxCompiler),
+        react.overrideDependencies({
+          dependencies: {
+            '@teambit/mdx.ui.mdx-scope-context': '0.0.368',
+            '@mdx-js/react': '1.6.22',
+          },
+        }),
+        react.overrideCompilerTasks([compiler.createTask('MDXCompiler', mdxCompiler)]),
+      ]
+
+      const mdxComposedEnv: MdxEnv = envs.merge<MdxEnv, ReactEnv>(
+        new MdxEnv(),
+        envs.compose(react.reactEnv, envOverrides)
+      );
```

Comment:
I think it will be cleaner with:
```js
const aspectEnv = react.compose(
      envOverrides,
      new MdxEnv()
    );
```
Just make sure it works

Reply:
Note that i changed MdxEnv to not just be a reactEnv. You think it'll still work?

Reply:
yes, I think so. it is basically the same as what you did in a cleaner way

Reply:
done